ui/src/literature/components/CitationSummarySwitch.jsx

```diff
@@ -1,23 +1,44 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { BarChartOutlined } from '@ant-design/icons';
 import { Switch, Tooltip } from 'antd';
+import PropTypes from 'prop-types';
+import { WITH_CITATION_SUMMARY } from '../constants';
 
 const CHART_ICON = <BarChartOutlined />;
 
-function CitationSummarySwitch(props) {
-  const { checked } = props;
-
+function CitationSummarySwitch({
+  checked,
+  updateHash,
+  onChange,
+  isCitationSummaryEnabled,
+}) {
+  useEffect(
+    () => {
+      if (isCitationSummaryEnabled) {
```

Comment:
you can pull this logic out the container and have something like `onCitationSummaryUserPreferenceChange` (yeah kinda long name 😄)

Reply:
and we will eventually get rid of this effect, when we find a better way.